src/lib/y2network/interfaces_collection.rb

```diff
@@ -87,5 +96,70 @@ def ==(other)
     end
 
     alias_method :eql?, :==
+
+    # Returns the interfaces that are enslaved in the given bridge
+    #
+    # @param master [String] bridge name
+    # @return [Array<String>] a list of interface names
+    # TODO: move to class of interface type br
+    def bridge_slaves(master)
+      bridge_index.select { |_k, v| v == master }.keys
+    end
+
+    # Creates list of devices enslaved in the bond device.
+    #
+    # @param bond_iface [String] a name of an interface of bond type
+    # @return list of names of interfaces enslaved in the bond_iface
+    # TODO: move to class of interface type bond
+    def bond_slaves(bond_iface)
+      bond_map = Yast::NetworkInterfaces::FilterDevices("netcard").fetch("bond", {}).fetch(bond_iface, {})
```

Comment:
`Y2Network::InterfacesCollection` is a class which holds a set of interfaces and allows you to query the collection (filtering by type, name, whatever). And that is its only responsibility.

So IMHO, coupling it with `NetworkInterfaces` is wrong. If you need to get the slaves for a given interface, you should be able to do so by querying the interfaces included in the collection. Otherwise, I fear that the logic does not belong to this class.

Reply:
i'm moving old functionality to new place ... if i change everything I would like to ... and other things you would like ... I'll get another huge PR which won't get accepted.

I planned to create Interface class descendants for bond and br as a next step and clean this methods which should be moved into these new classes when moving.